contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,393 @@
+  if (request.nextUrl.pathname=== '/test') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  return NextResponse.next()
+}
+
+export const config = {
+  matcher: ['/test', '/control'],
+};
+```
+
+### Getting or creating a user ID for flag evaluation
+
+To evaluate the flags for the experiment in our middleware, we need a distinct user ID to target. We want the user experience to be consistent so we:
+
+1. Check if a `distinct_id` exists in the PostHog cookie, and use it if so.
+2. Create a `distinct_id` and bootstrap it to the client if not.
+
+This requires using your project API key to get the cookies, parsing them as JSON, and potentially creating a distinct ID using `crypto.randomUUID()`. Altogether, this looks like this:
+
+```js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  const ph_project_api_key = '<ph_project_api_key>'
+  const ph_cookie_key = `ph_${ph_project_api_key}_posthog`
+  const cookie = request.cookies.get(ph_cookie_key);
+
+  let distinct_id;
+  if (cookie) {
+    // Use PostHog distinct_id 
+    distinct_id = JSON.parse(cookie.value).distinct_id;
+  } else {
+    // Create new distinct_id
+    distinct_id = crypto.randomUUID();
+  }
+//... rest of code
+```
+
+### Evaluating our redirect test with PostHog
+
+With our distinct ID, we can use the PostHog API to evaluate the `main-redirect` feature flag we set up earlier (because we can’t use PostHog SDKs in Next.js middleware). 
+
+We evaluate the flag by making a POST request to the `https://app.posthog.com/decide?v=3` route (replace `app` with `eu` if you’re on EU Cloud) with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, this looks like this:
+
+```js
+//... rest of code
+
+	const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      api_key: ph_project_api_key,
+      distinct_id: distinct_id
+    })
+  };
+  
+  // Evaluate experiment flag
+  const ph_request = await fetch(
+    'https://app.posthog.com/decide?v=3', // or eu
+    requestOptions
+  );
+  const data = await ph_request.json();
+  const flagResponse = data.featureFlags['main-redirect']
+
+  // Redirect to correct page
+  if (request.nextUrl.pathname=== '/test' && flagResponse === 'control') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  if (request.nextUrl.pathname=== '/control' && flagResponse === 'test') {
+    return NextResponse.redirect(new URL('/test', request.url))
+  }
+  return NextResponse.next()
+}
+//... rest of code
+```
+
+### Capturing exposure
+
+To get accurate results for our experiment, we also need to capture a `$feature_flag_called` exposure event after the feature flag has been evaluated, but before redirecting. This requires another POST request like this:
```

Comment:
What is an exposure and exposure event? Why is it important? (I know what it is, but I dont think a new reader will). Let's explain here what we're doing and why

Reply:
rewrote